onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py

```diff
@@ -902,109 +894,110 @@ def main():
-
-                for fp_path, int4_path in zip(old_paths, new_paths):
-                    if os.path.exists(fp_path):
-                        model = onnx.load_model(fp_path, load_external_data=True)
-                        quant = MatMul4BitsQuantizer(
-                            model=model,
-                            block_size=args.block_size,
-                            is_symmetric=True,
-                            accuracy_level=args.int4_accuracy_level,
-                            nodes_to_exclude=[],
-                        )
-                        quant.process()
-                        quant.model.save_model_to_file(int4_path, use_external_data_format=True)
-                        del model
-                        del quant
-                        logger.info(f"The ONNX model at {fp_path} has been quantized to int4 and saved at {int4_path}!")
-                        remove_existing_model(fp_path)
+                elif args.precision == Precision.INT4:
+                    if args.execution_provider != "cpu":
+                        old_paths = convert_to_float16(args, old_paths, rank)
+
+                    decoder_model_int4_path = os.path.join(
+                        args.output, f"rank_{rank}_{args.model_name}_decoder_model_int4.onnx"
+                    )
+                    decoder_with_past_model_int4_path = os.path.join(
+                        args.output, f"rank_{rank}_{args.model_name}_decoder_with_past_model_int4.onnx"
+                    )
+                    decoder_merged_model_int4_path = os.path.join(
+                        args.output, f"rank_{rank}_{args.model_name}_decoder_merged_model_int4.onnx"
+                    )
+                    new_paths = [decoder_model_int4_path, decoder_with_past_model_int4_path, decoder_merged_model_int4_path]
+
+                    for fp_path, int4_path in zip(old_paths, new_paths):
+                        if os.path.exists(fp_path):
+                            model = onnx.load_model(fp_path, load_external_data=True)
+                            quant = MatMul4BitsQuantizer(
+                                model=model,
+                                block_size=args.block_size,
+                                is_symmetric=True,
+                                accuracy_level=args.int4_accuracy_level,
+                                nodes_to_exclude=[],
+                            )
+                            quant.process()
+                            quant.model.save_model_to_file(int4_path, use_external_data_format=True)
+                            del model
+                            del quant
+                            logger.info(f"The ONNX model at {fp_path} has been quantized to int4 and saved at {int4_path}!")
+                            remove_existing_model(fp_path)
         barrier()
 
     logger.info("Verifying parity on all ONNX models created")
```

Comment:
The parity checker will currently fail on the Dynamo exported model since its input names are different than the expected names. Let's return early so we don't run it.

```python
# Return early if using Dynamo exporter
if args.use_dynamo_export:
    return

# Rest of parity checking code for TorchScript exported model stays here
logger.info("Verifying parity on all ONNX models created")
...
```

Reply:
thank you